apps/astarte_trigger_engine/lib/astarte_trigger_engine/events_consumer.ex

```diff
@@ -248,27 +245,38 @@ defmodule Astarte.TriggerEngine.EventsConsumer do
   end
 
   defp retrieve_trigger_configuration(realm_name, trigger_id) do
-    query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(
-        "SELECT value FROM kv_store WHERE group='triggers' AND key=:trigger_id;"
-      )
-      |> DatabaseQuery.put(:trigger_id, trigger_id)
+    statement =
+      "SELECT value FROM #{realm_name}.kv_store WHERE group='triggers' AND key=:trigger_id;"
+
+    params = %{"trigger_id" => trigger_id}
 
-    with {:ok, client} <- Database.connect(realm: realm_name),
-         {:ok, result} <- DatabaseQuery.call(client, query),
-         [value: trigger_data] <- DatabaseResult.head(result),
+    with :ok <- check_realm_name_ok(realm_name),
```

Comment:
The case of an invalid realm name could be handled explicitly rather than returning `{:error, :trigger_not_found}` which can be misleading